src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -781,7 +781,7 @@ protected function redirectAfterSave($isNewRecord)
         $controller = $this->getToplevelController();
         if ($isNewRecord) {
             return $controller->redirect($this->Link());
-        } elseif ($this->gridField->getList()->byID($this->record->ID)) {
+        } elseif ($this->gridField->getList()->filter('ID', $this->record->ID)->exists()) {
```

Comment:
While not strictly reducing the _number_ of queries, I've updated a bunch of calls to `find()` and `byID()` to instead use `filter()->exists()` which is more efficient.